### PR TITLE
[ENH] Reader for .xyz files

### PR DESCRIPTION
**Pull request by ctk3b**
Super low priority but one step closer to "Read and write from every MD format imaginable".

### CODE REVIEW

File: mdtraj/core/trajectory.py [hunk 1]
```diff
@@ -1236,6 +1238,19 @@ def save_lammpstrj(self, filename, force_overwrite=True):
                     cell_angles=self.unitcell_angles,
                     cell_lengths=self.unitcell_lengths)
 
+    def save_xyz(self, filename, force_overwrite=True):
+        """Save trajectory to LAMMPS custom dump format
```
Comment (rmcgibbo) [inline mdtraj/core/trajectory.py]: LAMMPS -> xyz?

File: mdtraj/core/trajectory.py [hunk 2]
```diff
@@ -1236,6 +1238,19 @@ def save_lammpstrj(self, filename, force_overwrite=True):
...
+
+        Parameters
+        ----------
+        filename : str
+            filesystem path in which to save the trajectory
+        force_overwrite : bool, default=True
+            Overwrite anything that exists at filename, if its already there
+        """
+        with XYZTrajectoryFile(filename, 'w', force_overwrite=True) as f:
```
Comment (rmcgibbo) [inline mdtraj/core/trajectory.py]: pass through the `force_overwrite` option, instead of fixing to true?

File: mdtraj/formats/xyzfile.py [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+            types = ['X' for _ in xrange(xyz.shape[1])]
+
+        in_units_of(xyz, 'nanometers', self.distance_unit, inplace=True)
+
+        for i in range(xyz.shape[0]):
+            self._fh.write('{0}\n'.format(xyz.shape[1]))
+            self._fh.write("Created with MDTraj {0}, {1}".format(version, str(date.today())))
+
+            for j, coord in enumerate(xyz[i]):
+                self._fh.write('{0} {1:8.3f} {2:8.3f} {3:8.3f}\n'.format(
```
Comment (rmcgibbo) [inline mdtraj/formats/xyzfile.py]: is there are particular standard documented somewhere that this format adheres to that we can link to that specifies things like the level of precision?

Comment (ctk3b) [inline mdtraj/formats/xyzfile.py]: From my understanding there is no universal standard but this is the one that VMD uses. I've updated the doc string to reference this fact.

**Comment by rmcgibbo** [comment]
I know they're a little bit repetitive, but can you also add the XYZTrajectoryFile to the appropriate tests in https://github.com/mdtraj/mdtraj/blob/master/mdtraj/tests/test_trajectory.py, like `test_seek_read_mode` and maybe the list of files [here](https://github.com/mdtraj/mdtraj/blob/master/mdtraj/tests/test_trajectory.py#L168-L169)

**Comment by rmcgibbo** [comment]
This looks awesome!

**Pull request merged**